### PR TITLE
Build and push images to ghcr.io, sign images with cosign.

### PR DESCRIPTION
**Pull request by janisz**
First part of moving building + pushing images from circle-CI to github actions.

Adds the following:
- Build and push images to GitHub Container Registry via GitHub Actions
- Sign built images of kube-linter via cosign (keyless + with public key)
- Add documentation on how to verify kube-linter docker image signatures

### CODE REVIEW

**Comment by janisz** [comment]
Let's get #296 in and then this

**Comment by janisz** [comment]
Let's get #296 in and then this

**Review by dhaus67 — APPROVED** [review]
LGTM let's see after first merge run which imag tags built and whether there's any need for changes before we add dockerhub

**Pull request merged**
